subgraph.template.yaml

```diff
@@ -472,25 +478,28 @@ dataSources:
       eventHandlers:
         - event: WithdrawalInitiated(address,indexed address,indexed address,indexed uint256,uint256,uint256)
           handler: handleWithdrawalInitiated
-  # - kind: ethereum/contract
-  #   name: BlockHandler
-  #   network: {{network}}
-  #   source:
-  #     address: "{{epochManager}}"
-  #     abi: EpochManager
-  #     startBlock: {{blockNumber}}
-  #   mapping:
-  #     kind: ethereum/events
-  #     apiVersion: 0.0.5
-  #     language: wasm/assemblyscript
-  #     file: ./src/mappings/block.ts
-  #     entities:
-  #       - GraphNetwork
-  #     abis:
-  #       - name: EpochManager
-  #         file: ./node_modules/@graphprotocol/contracts/dist/abis/EpochManager.json
-  #     blockHandlers:
-  #       - handler: handleBlock
+        - event: DepositFinalized(indexed address,indexed address,indexed address,uint256)
+          handler: handleDepositFinalized
+  - kind: ethereum/contract
+    name: ArbRetryableTx
+    network: {{network}}
+    source:
+      abi: ArbRetryableTx
+      address: "0x000000000000000000000000000000000000006E"
+      startBlock: {{bridgeBlockNumber}}
+    mapping:
+      kind: ethereum/events
+      apiVersion: 0.0.7
+      language: wasm/assemblyscript
+      file: ./src/mappings/arbRetryableTx.ts
+      entities:
+        - GraphNetwork
+      abis:
+        - name: ArbRetryableTx
+          file: ./abis/ArbRetryableTx.json
+      eventHandlers:
+        - event: RedeemScheduled(indexed bytes32,indexed bytes32,indexed uint64,uint64,address,uint256,uint256)
+          handler: handleRedeemScheduled
```

Comment:
Is this address correct? and if so, should it also be hardcoded for all L1s?

Reply:
Yes it's correct! It's an Arbitrum pre-compile, meaning it's a "contract" baked in the arbitrum node, always has the same address regardless of the environment.

Reply:
If you are curious here you can read more about them: https://developer.arbitrum.io/arbos/common-precompiles